### PR TITLE
feat: prune button for pods

### PR DESCRIPTION
**Pull request by cdrage**
feat: prune button for pods

### What does this PR do?

Adds a prune button for pods.

### Screenshot/screencast of this PR

https://user-images.githubusercontent.com/6422176/219142703-e5bbae77-beb1-4f17-a353-84f6c1bf74fd.mov

### What issues does this PR fix or reference?

Reference: containers#924

### How to test this PR?

`yarn watch`, press the "prune" button in pods

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: packages/renderer/src/lib/engine/Prune.svelte [hunk 1]
```diff
@@ -32,16 +32,20 @@ async function prune(type: string) {
           console.error(error);
         }
       });
-
+      break;
+    case 'pods':
+      engines.forEach(async engine => {
+        try {
+          await window.prunePods(engine.id);
```
Comment (jeffmaury) [inline packages/renderer/src/lib/engine/Prune.svelte:39]: We should probably not wait for the result for better parallelism as there is no feedback

**Comment by cdrage** [comment]
Converting to draft so I can:
- [x] Dont show button on no pods
- [x] Remove await

**Comment by cdrage** [comment]
Ready for review 👍

**Review by benoitf — APPROVED** [review]
it does the job but maybe could be nice (not for this PR) that we're able to select for which engine we do that

File: packages/renderer/src/lib/pod/PodsList.svelte [hunk 2]
```diff
@@ -194,6 +209,9 @@ function errorCallback(pod: PodInfoUI, errorMessage: string): void {
   title="pods"
   subtitle="Hover over an pod to view action buttons; click to open up full details.">
   <div slot="additional-actions" class="space-x-2 flex flex-nowrap">
+    {#if $filtered.length > 0}
```
Comment (benoitf) [inline packages/renderer/src/lib/pod/PodsList.svelte]: as discussed on the containers item, probably we need to check the presence of pods not the filtered one

**Comment by cdrage** [comment]
@benoitf Updated to use podsInfos instead

File: packages/renderer/src/lib/engine/EngineInfoUI.ts [hunk 3]
```diff
@@ -20,5 +20,5 @@
 export interface EngineInfoUI {
   id: string;
   name: string;
-  type: 'podman' | 'docker';
+  type?: 'podman' | 'docker';
```
Comment (benoitf) [inline packages/renderer/src/lib/engine/EngineInfoUI.ts]: do we have cases where it's undefined ?

Comment (cdrage) [reply packages/renderer/src/lib/engine/EngineInfoUI.ts]: You're right. Should be removed, I've opened up PR: https://github.com/containers/podman-desktop/pull/1535 and let's rebase against that once it's merged.

**Comment by cdrage** [comment]
Rebased (conflict) and ready for another review 💯

**Review by benoitf — APPROVED** [review]
approving again

**Pull request merged**